auth_ident/preprocessing/save_hdf_codenet.py

```diff
@@ -0,0 +1,217 @@
+    :param keep_repeats (boolean): Keep multiple problem submissions.
+    :param languages (list of strings): Keep files with these languages.
+    :param val_test_split (float): Fraction of files to set aside for
+    each of validation and testing.
+    :param min_file_size (int): Minimum file size to keep (in bytes).
+    :param dataset (string): Restrict to single dataset AIZU or AtCoder.
+
+    """
+    submissions = []
+    codenet_root = os.path.normpath(codenet_root)
+    metadata_root = os.path.join(codenet_root, 'metadata')
+    data_root = os.path.join(codenet_root, 'data')
+
+    if not keep_repeats:
+        dups = set()
+        for language in languages:
+            dups.update(get_dups(codenet_root, language))
+
+    problems_df = pd.read_csv(os.path.join(metadata_root,
+                                           'problem_list.csv'))
+    if dataset:
+        prob_ids = list(
+            problems_df.loc[problems_df['dataset'] == dataset]['id'])
+    else:
+        prob_ids = list(problems_df['id'])
+
+    prob_csvs = [prob_id + '.csv' for prob_id in prob_ids]
+
+    total_checked = 0
+    size_rejected = 0
+    dup_rejected = 0
+    for prob_num, csv_file in enumerate(prob_csvs):
+
+        print("Processing {} {}/{} ({} files)".format(csv_file,
+                                                      prob_num,
+                                                      len(prob_csvs),
+                                                      len(submissions)))
+        metadata_df = pd.read_csv(os.path.join(metadata_root, csv_file))
+
+        for language in languages:
+            language_df = metadata_df.loc[metadata_df['language'] == language]
+            for _, frame in language_df.iterrows():
+                total_checked += 1
+                if frame['code_size'] <= min_file_size:
+                    size_rejected += 1
+                    continue
+                if not keep_repeats and frame['submission_id'] in dups:
+                    dup_rejected += 1
+                    continue
+
+                file_name = (frame['submission_id'] + "."
+                             + frame['filename_ext'])
+                full_path = os.path.join(data_root,
+                                         frame['problem_id'],
+                                         language, file_name)
+                with open(full_path, 'rb') as content_file:
+                    contents = content_file.read()
+                    submissions.append((frame['problem_id'],
+                                        frame['user_id'],
+                                        language,
+                                        contents))
+
+    frame = pd.DataFrame({"problem_id": [v[0] for v in submissions],
+                          "user_id": [v[1] for v in submissions],
+                          "language": [v[2] for v in submissions],
+                          "file_content": [v[3] for v in submissions]})
```

Comment:
Could we add the filepath to the dataframe? Just put in the full_path for each file? I just need the file path to feed into the ibm tokenizer.

Reply:
Should be ready to go